File: app/src/main/java/org/techtown/gtguildraid/adapters/MemberCardAdapter.java
package org.techtown.gtguildraid.adapters;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import org.techtown.gtguildraid.R;
import org.techtown.gtguildraid.models.GuildMember;

import java.util.List;

public class MemberCardAdapter extends RecyclerView.Adapter<MemberCardAdapter.ViewHolder> {
    private List<GuildMember> memberList;

    public MemberCardAdapter(List<GuildMember> memberList){
        this.memberList = memberList;
        notifyDataSetChanged();
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.card_member, parent, false);

        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull final ViewHolder holder, int position) {
        GuildMember member = memberList.get(position);
        holder.setItem(member);
    }

    @Override
    public int getItemCount() {
        return memberList.size();
    }

    static class ViewHolder extends RecyclerView.ViewHolder{
        TextView nickname;
        TextView remark;
        public ViewHolder(@NonNull View itemView) {
            super(itemView);

            nickname = itemView.findViewById(R.id.nickname);
            remark = itemView.findViewById(R.id.remark);
        }

        public void setItem(GuildMember member){
            nickname.setText(member.getName());
            remark.setText(member.getRemark());
        }
    }
}

File: app/src/main/java/org/techtown/gtguildraid/member/MemberCardAdapter.kt
package org.techtown.gtguildraid.member

import org.techtown.gtguildraid.models.entities.GuildMember
import androidx.recyclerview.widget.RecyclerView
import android.view.ViewGroup
import android.view.LayoutInflater
import android.view.View
import org.techtown.gtguildraid.R
import android.widget.TextView

class MemberCardAdapter(private val memberList: List<GuildMember?>?) :
    RecyclerView.Adapter<MemberCardAdapter.ViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.card_member, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val member = memberList?.get(position)
        if (member != null) {
            holder.setItem(member)
        }
    }

    override fun getItemCount(): Int {
        return memberList?.size ?: 0
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var nickname: TextView
        var remark: TextView
        fun setItem(member: GuildMember) {
            nickname.text = member.name
            remark.text = member.remark
        }

        init {
            nickname = itemView.findViewById(R.id.nickname)
            remark = itemView.findViewById(R.id.remark)
        }
    }

    init {
        notifyDataSetChanged()
    }
}